packaging/Makefile_packaging.mk

```diff
@@ -0,0 +1,516 @@
+DEB_TARBASE := $(DEB_TOP)/$(DEB_NAME)_$(DEB_VERS)
+SOURCES := $(addprefix _topdir/SOURCES/,$(notdir $(SOURCE)) $(PATCHES))
+ifeq ($(ID_LIKE),debian)
+DEBS    := $(addsuffix _$(DEB_VERS)-1_amd64.deb,$(shell sed -n '/-udeb/b; s,^Package:[[:blank:]],$(DEB_TOP)/,p' debian/control))
+DEB_PREV_RELEASE := $(shell dpkg-parsechangelog -S version)
+DEB_DSC := $(DEB_NAME)_$(DEB_PREV_RELEASE)$(GIT_INFO).dsc
+#Ubuntu Containers do not set a UTF-8 environment by default.
+ifndef LANG
+export LANG = C.UTF-8
+endif
+ifndef LC_ALL
+export LC_ALL = C.UTF-8
+endif
+TARGETS := $(DEBS)
+else
+# CentOS/Suse packages that want a locale set need this.
+ifndef LANG
+export LANG = en_US.utf8
+endif
+ifndef LC_ALL
+export LC_ALL = en_US.utf8
+endif
+TARGETS := $(RPMS) $(SRPM)
+endif
+
+define install_repos
+	for repo in $($(DISTRO_BASE)_PR_REPOS)                              \
+	            $(PR_REPOS) $(1); do                                    \
+	    branch="master";                                                \
+	    build_number="lastSuccessfulBuild";                             \
+	    if [[ $$repo = *@* ]]; then                                     \
+	        branch="$${repo#*@}";                                       \
+	        repo="$${repo%@*}";                                         \
+	        if [[ $$branch = *:* ]]; then                               \
+	            build_number="$${branch#*:}";                           \
+	            branch="$${branch%:*}";                                 \
+	        fi;                                                         \
+	    fi;                                                             \
+	    case $(DISTRO_ID) in                                            \
+	        el7) distro="centos7";                                      \
+	        ;;                                                          \
+	        sle12.3) distro="sles12.3";                                 \
+	        ;;                                                          \
+	        sl42.3) distro="leap42.3";                                  \
+	        ;;                                                          \
+	        sl15.1) distro="leap15.1";                                  \
+	        ;;                                                          \
+	    esac;                                                           \
+	    baseurl=$${JENKINS_URL:-https://build.hpdd.intel.com/}job/daos-stack/job/$$repo/job/$$branch/; \
+	    baseurl+=$$build_number/artifact/artifacts/$$distro/;           \
+	    $(call install_repo,$$baseurl);                                 \
+        done
+endef
+
+all: $(TARGETS)
+
+%/:
+	mkdir -p $@
+
+%.gz: %
+	rm -f $@
+	gzip $<
+
+_topdir/SOURCES/%: % | _topdir/SOURCES/
+	rm -f $@
+	ln $< $@
+
+# At least one spec file, SLURM (sles), has a different version for the
+# download file than the version in the spec file.
+ifeq ($(DL_VERSION),)
+DL_VERSION = $(VERSION)
+endif
+
+ifneq ($(dir $(SOURCE)),./)
```

Comment:
Why does this recipe need this conditional?  An `.asc` target should only be specified by a project that is packaging something that supplies one, otherwise this recipe is dormant anyway.

Reply:
This is to prevent it attempting to use curl to refresh / replace the local copy every time the spec file or calling makefile is updated.

Reply:
I think it would be better to understand why `make` thinks it needs updating.  This kind of band-aid around `make` doing what it's supposed to (when it's given the right rules) tends to just cause more unexpected situations.

Reply:
Before moving the rule in Makefile-rpm.mk to be after the include packaging/Makefile_packaging.mk, one of these rules was replacing the rule in Makefile-rpm.mk and attempted to be executed, which failed as the path was not a URL.